controllers/submariner/gateway_resources.go

```diff
@@ -252,8 +254,12 @@ func (r *Reconciler) reconcileGatewayDaemonSet(
 		return nil, err
 	}
 
-	err = metrics.Setup(instance.Namespace, instance, daemonSet.GetLabels(), gatewayMetricsServerPort, r.config.ScopedClient,
-		r.config.RestConfig, r.config.Scheme, reqLogger)
+	metricsSvcLabels := map[string]string{
+		"app": names.MetricsProxyComponent,
+	}
```

Comment:
It would be easier (and more straightforward) to refactor `Setup` to receive the app, as it's the only thing it cares about in the labels being passed to it.
That would obviate the need for this code (and the similar change for GN)

Reply:
Agree. Wasn't sure if we need any other labels. Wanted to refactor but went this this first to keep changes to minimum and then discuss it in review.